src/cdk/schematics/update-tool/file-system.ts

```diff
@@ -43,8 +43,10 @@ export interface DirectoryEntry {
  * changes. This is necessary to support virtual file systems as used in the CLI devkit.
  */
 export abstract class FileSystem {
-  /** Checks whether the given file or directory exists. */
-  abstract exists(path: WorkspacePath): boolean;
+  /** Checks whether the given file exists. */
+  abstract existsFile(path: WorkspacePath): boolean;
+  /** Checks whether the given directory exists. */
+  abstract existsDirectory(path: WorkspacePath): boolean;
```

Comment:
nit: the method names here sound a little weird to me. Maybe they should be `fileExists` and `directoryExists`?